components/builder-api/src/server/framework/middleware.rs

```diff
@@ -98,12 +99,56 @@ fn authenticate(req: &HttpRequest<AppState>, token: &str) -> error::Result<Sessi
 
     // Check for a valid personal access token
     if bldr_core::access_token::is_access_token(token) {
-        let session =
-            bldr_core::access_token::validate_access_token(&req.state().config.api.key_path, token)
-                .map_err(|_| NetError::new(ErrCode::BAD_TOKEN, "net:auth:bad-token"))?;
-        revocation_check(req, session.get_id(), token)
-            .map_err(|_| NetError::new(ErrCode::BAD_TOKEN, "net:auth:revoked-token"))?;
-        return Ok(session);
+        let mut memcache = req.state().memcache.borrow_mut();
+        match memcache.get_session(token) {
+            Some(session) => {
+                trace!("Session {} Cache Hit!", token);
+                return Ok(session);
+            }
+            None => {
+                trace!("Session {} Cache Miss!", token);
+                // Pull the session out of the current token provided so we can validate
+                // it against the db's tokens
+                let session = bldr_core::access_token::validate_access_token(
+                    &req.state().config.api.key_path,
+                    token,
+                ).map_err(|_| NetError::new(ErrCode::BAD_TOKEN, "net:auth:bad-token"))?;
+
+                // If we can't find a token in the cache, we need to round-trip to the
+                // db to see if we have a valid session token. The first time this is used,
+                // it is only to get the account_id so we can do the lookup
```

Comment:
This comment isn't clear - the account_id is already available as the session id..